lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,436 @@
+  const _RichTextFieldModal({required this.title, required this.controller, required this.placeholder, this.focus});
+
+  final Widget title;
+  final quill.QuillController controller;
+  final String placeholder;
+  final FocusNode? focus;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+
+    final focusNode = focus ?? useFocusNode();
+
+    final isFocused = useState(focusNode.hasFocus);
+    final hasSelection = useState(controller.selection.start != controller.selection.end);
+
+    useEffect(() {
+      void focusUpdate() => isFocused.value = focusNode.hasFocus;
+      void selectionUpdate() => hasSelection.value = controller.selection.start != controller.selection.end;
+
+      focusNode.addListener(focusUpdate);
+      controller.addListener(selectionUpdate);
+
+      return () {
+        focusNode.removeListener(focusUpdate);
+        controller.removeListener(selectionUpdate);
+      };
+    });
+
+    final editor = _ThemedEditor(
+      controller: controller,
+      placeholder: placeholder,
+      backgroundColor: theme.neutralSwatch.shade800,
+      focus: focusNode,
+    );
+
+    final okButton = CustomTextButton(
+      text: strings.ok.toUpperCase(),
+      onPressed: Navigator.of(context).pop,
+    );
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              Align(alignment: Alignment.centerRight, child: okButton),
+              context.verticalBox(Spacing.xxxSmall),
+              title,
+              context.verticalBox(Spacing.xxSmall),
+              Expanded(child: editor),
+              context.verticalBox(Spacing.xxSmall),
+            ],
+          ).withSymmetricalPadding(context, horizontal: Spacing.medium),
+        ),
+        // Only show toolbar action items when the field has focus or when the user has an active selection.
+        if (isFocused.value || hasSelection.value) ...[
+          _RichTextFieldToolbar(controller),
+          SizedBox(height: MediaQuery.of(context).viewInsets.bottom),
+        ]
+      ],
+    );
+  }
+}
+
+/// Provides a custom styled and reusable [quill.QuillEditor].
+///
+/// Ideally this wrapper would not be necessary but `quill` forces us to provide a set of attributes when building a
+/// [quill.QuillEditor] that are shared between this file editor instances.
+///
+/// It basically reduces boilerplate.
+class _ThemedEditor extends ConsumerWidget {
+  const _ThemedEditor({
+    required this.controller,
+    required this.backgroundColor,
+    required this.placeholder,
+    this.focus,
+    this.showCursor = true,
+    this.readonly = false,
+  });
+
+  final quill.QuillController controller;
+  final Color backgroundColor;
+  final String placeholder;
+  final FocusNode? focus;
+  final bool showCursor;
+  final bool readonly;
```

Comment:
can be explained

Reply:
Updated docs and removed `showCursor` in favor of using `readonly`